inc/sainexthopgroup.h

```diff
@@ -311,6 +327,50 @@ typedef enum _sai_next_hop_group_member_attr_t
 
 } sai_next_hop_group_member_attr_t;
 
+typedef enum _sai_next_hop_group_map_type_t
+{
+    /** Next hop group map forwarding-class to index */
+    SAI_NEXT_HOP_GROUP_MAP_TYPE_FORWARDING_CLASS_TO_INDEX
+
+} sai_next_hop_group_map_type_t;
+
+typedef enum _sai_next_hop_group_map_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_NEXT_HOP_GROUP_MAP_ATTR_START,
+
+    /**
+     * @brief Next hop group map type
+     *
+     * @type sai_next_hop_group_map_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
```

Comment:
`@isresourcetype true` markdown needs to be added in order to query the switch's object availability.

Reply:
I think it can be a resource regardless, as the ``@isresourcetype`` flag is to flag a specific attribute as separating the object into multiple resource pools.

I'm not sure what types could be added in the future, but I can add that tag if we think it is probable to have different resource pools for those on some implementations.

Reply:
I'm not 100% sure, but while writing some swss code which makes use of the NHG maps, I couldn't query the switch object availability for NHG maps until I've added that flag. The SAI metadata validation was throwing an error saying that the queried object was not a resource type.